cmip6_downscaling/methods/bcsd.py

```diff
@@ -264,44 +378,57 @@ def return_x_train_full_time(
         Date for prediction period end (e.g. '2090')
     variable: str
         The variable included in the dataset.
+    latmin : str
+            From run hyperparameters
+    latmax : str
+            From run hyperparameters
+    lonmin : str
+            From run hyperparameters
+    lonmax : str
+            From run hyperparameters
 
     Returns
     -------
     xr.Dataset
         x_train rechunked dataset in full time.
     """
-    X_train = load_cmip(source_ids=gcm, variable_ids=[variable], return_type='xr').sel(
-        time=slice(train_period_start, train_period_end)
+    gcm_train_ds = load_cmip(source_ids=gcm, variable_ids=[variable], return_type='xr')
+    gcm_train_ds_subset = subset_dataset(
+        gcm_train_ds, train_period_start, train_period_end, latmin, latmax, lonmin, lonmax
     )
-    X_train['time'] = y_full_time_ds.time.values
 
-    X_train_full_time_ds, X_train_full_time_path = rechunk_zarr_array(
-        X_train,
-        zarr_array_location=None,
-        variable=variable,
-        chunk_dims=('lat', 'lon'),
+    # Q: Ask Ori why this exists
+    gcm_train_ds_subset['time'] = coarse_obs_full_time_ds.time.values
```

Comment:
this call was to force the timestamps for the cmip data to use the friendlier era5 timestamps. (i forget which dataset used which time formats). i could picture this introducing a tricky bug though (for instance if gcm timestamp didn't align for some reason) so we could use another conversion system if that is better.

Reply:
or maybe just add a datetime equivalency test or something